### PR TITLE
Add storybook for HierarchicalLocationsView, remove hardcoded theme colors

### PR DESCRIPTION
**Pull request by sqs**

### CODE REVIEW

**Review by felixfbecker — APPROVED** [review]
[`CodeSnippet`](https://github.com/sourcegraph/sourcegraph/blob/main/client/branded/src/components/CodeSnippet.scss) should be updated too (SCSS file can get deleted, TSX can use `bg-code`)

File: client/shared/src/util/url.ts [hunk 1]
```diff
@@ -169,7 +169,7 @@ const parsePosition = (string: string): Position => {
  */
 export function parseRepoURI(uri: RepoURI): ParsedRepoURI {
     const parsed = new URL(uri)
-    const repoName = parsed.hostname + parsed.pathname
+    const repoName = parsed.hostname + parsed.pathname.replace(/^\/\//, '')
```
Comment (felixfbecker) [inline client/shared/src/util/url.ts:172]: **question**: Why is this needed? Where do those double slashes come from?

**Comment by felixfbecker** [comment]
Sorry approved the wrong PR

**Review by felixfbecker — CHANGES_REQUESTED** [review]

File: client/shared/src/util/url.ts (hunk 1 above)
Comment (sqs) [reply client/shared/src/util/url.ts:172]: Ah, I forgot to extract this and deal with it separately. I will remove it from this PR.

In storybooks, there is no `URL` polyfill, so it treats `git://` URIs as having a pathname of `//example.com/foo` and no hostname. In the main app, our `URL` polyfill (incorrectly per the spec) treats URIs of all schemes the same, so this is not needed. The reason I included this change here is that the storybook for this component would show the repo name as `//github.com/foo/bar`, which was weird. But this is a bigger change and should not be in this PR.

BTW, I think it would be good to completely remove these `git://` URIs from this part of our app if possible (and probably even everywhere). What do you think?

Comment (felixfbecker) [reply client/shared/src/util/url.ts]: We should just include all polyfills in Storybook - we already include e.g. `focus-visible`. Iirc we have a shared `polyfill.ts` file that we could just properly share with the webapp.

URL polyfill comes from babel-preset-env/core-js, which we share in Storybook too, so that should work.

> BTW, I think it would be good to completely remove these git:// URIs from this part of our app if possible (and probably even everywhere). What do you think?

I totally agree, these URIs are a pain everywhere they exist. I just never had the time to really rip them out. I think we're also gonna add alternative APIs to extensions next iteration so URI parsing is no longer needed, but even just not relying on them outside of the extension host would be great.

**Review by felixfbecker — APPROVED** [review]
